### ISSUE TITLE
Fetching single commit

### ISSUE THREAD

**Issue by hirokiosame** [isomorphic-git/isomorphic-git#714]
Looking to be able to download a single specific commit.

Looks like git fetch is able to do that:
[`git fetch --depth=1 ../testrepo/.git $SHA1`](https://stackoverflow.com/questions/14872486/retrieve-specific-commit-from-a-remote-git-repository/30701724#30701724)

Is there a way to do this with isomorphic git?

**Comment by wmhilton** [comment]
Yup. Depends on whether the remote supports it (via `allow-tip-sha1-in-want` or `allow-reachable-sha1-in-want`) but I just tried this and it works:

```bash
mkdir tmp
cd tmp
isogit init
isogit clone --depth=1 --url=https://github.com/isomorphic-git/isomorphic-git --ref=9aa9c2aca7fc6086e0ced9fa886442beba7bcf79
```

**Comment by willstott101** [comment]
This breaks with `singleBranch: true` due to the way the `refs` map is built in the `_fetch` function for `updateRemoteRefs`. But without `singleBranch`, whilst the command completes, the commit isn't found when I use `readCommit`. It seems to essentially fallback to `fetch --all` without `singleBranch`.
